### PR TITLE
interface for all

### PR DESCRIPTION
**Pull request by AmaurX**
## Description

- Made the launch file modular using existing `robot_type` argument.
- Added the argument to the actual roslaunch command in launch.sh
- For the config files of the camera node:
    - Created the one for watchtower and traffic_light
    - Renamed the default one to duckiebot.yaml
    - the goal is to use directly `robot_type` as the config file name. Avoids having to change the actual param_file_name for the whole `all_drivers.launch`, because then we would need a duplicate watchtower.yaml in every package

- To launch on a watchtower, add `-e ROBOT_TYPE=watchtower` to the command line

- This shouldn't break anything, it was tested 

## Type of change

Please mark the relevant options with X:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change is constrained only to one package, any interfaces to nodes from other packages are maintained
- [ ] This change is constrained only to packages in this repository, any interfaces to nodes from other repositories are maintained
- [ ] This change required a documentation update
- [ ] This change is only a documentation update
- [ ] This change restructures the repository (at level higher than package level)
- [ ] This change requires a change in the Duckiebook (if yes, please describe and, if possible, make a pull request for an edit in the book and link it here)

## Tests

Please describe all tests ran and their outcome. It is highly recommended that you at least do the tests listed by default, even if you only changed the documentation.

- [X] Made the Duckiebot move with keyboard control
- [X] Visually checked the published camera stream
- [X] Verified that the lane following demo works
- [X] Verified that the indefinite navigation demo works

## Style compliance
Please make sure that all your changes conform to the Duckietown (code style)[link] and code (documentation style)[link] guidelines:

- [X] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [X] I checked that this pull request is compliant with the Duckietown code style
- [X] I checked that this pull request is compliant with the Duckietown code documentation style

## Next steps
Please describe if you have any questions, comments, or doubts towards the Duckietown software guardians regarding implementation, code style, running tests, documentation, etc.

## FOR THE REVIEWERS
__IMPORTANT:__ DO NOT CHANGE THIS SECTION!

- [x] Functionality (is the change necessary and is it implemented in the most appropriate way): @liampaull or @afdaniele
- [x] Integration testing (verify the tests and run additional if necessary): @?
- [ ] Code style compliance: @gibernas
- [x] Documentation compliance: @aleksandarpetrov

_To the reviewers:_ If you deem necessary, request code edits, more tests, documentation update, or additional reviewing. __Do not review your own code!__

_PR Template, last edit 23 Aug 2019_

### CODE REVIEW

File: packages/duckiebot_interface/launch/all_drivers.launch [hunk 1]
```diff
@@ -3,64 +3,70 @@
...
-        <arg name="veh" value="$(arg veh)"/>
-        <arg name="param_file_name" value="$(arg param_file_name)"/>
+        <arg name="param_file_name" value="$(arg robot_type)"/>
     </include>
 
     <!-- LED Emitter: reads ROS messages and commands the LEDs -->
-    <include file="$(find led_emitter)/launch/led_emitter_node.launch">
-        <arg name="veh" value="$(arg veh)"/>
-    </include>
+    <group unless="$(eval arg('robot_type') == 'watchtower')">
```
Comment (AleksandarPetrov) [inline packages/duckiebot_interface/launch/all_drivers.launch]: @AmaurX why is this on for watchtower but not traffic light?

Comment (AmaurX) [reply packages/duckiebot_interface/launch/all_drivers.launch]: because the traffic light have leds

Comment (AmaurX) [reply packages/duckiebot_interface/launch/all_drivers.launch]: @AleksandarPetrov basically the default is to have leds, only the watchtowers don't have them

Comment (AmaurX) [reply packages/duckiebot_interface/launch/all_drivers.launch]: it's an `unless` statement. I think it is less dangerous that specifying each type that should have a led launch

Comment (afdaniele) [reply packages/duckiebot_interface/launch/all_drivers.launch]: @AmaurX did you know that starting from Kinetic, `$(eval EXP)` in roslaunch can take any valid python2 expression? So I guess you could try with OR as well.

**Pull request merged**